### PR TITLE
Fix touch input handling for glfw.

### LINKED ISSUE #20804: Incorrect TouchEvent mouse mapping for GLFW
Current behavior for GLFW TouchEvent handling:
- Each touchstart TouchEvent causes 2 mousedown events.
- Each touchend TouchEvent causes 2 mouseup events.
- touchmove events are ignored completely.

Expected behavior:

Until GLFW adds more fine-grained multitouch support, for a single primary touch:
- touchstart is mapped to  a single mousedown event.
- touchend is mapped to a single mouseup event.
- touchmove is mapped to a single mousemove event.

The reason mousedown/mouseup events are duplicated is that the GLFW callbacks for mouse events are added for touchstart/touchend, but the default browser behavior of emulating a mousedown/mouseup event for these touch events is **not** suppressed with a call to preventDefault in the event handler, resulting in a duplicate call to the mousedown/mouseup event handlers and the associated GLFW callebacks (note: touchmove does not trigger an emulated mouse event by default in the browser).

### PR DESCRIPTION
**Pull request by br0nk**
- Do not double-send mouse events for touches.
- Map touch move to mouse move for a single touch.
- Clean up mouse coordinate calculations.

Fixes #20804

### CODE REVIEW

**Comment by br0nk** [comment]
Ah crap, messed up my rebase.  Ignore the added commits.

File: src/library_glfw.js [hunk 1]
```diff
@@ -490,11 +514,51 @@ var LibraryGLFW = {
...
+          Browser.setMouseCoords(chosenTouch.pageX, chosenTouch.pageY);
+          Browser.setMouseCoords(chosenTouch.pageX, chosenTouch.pageY);
+          primaryChanged = true;
+        } else if (event.type === 'touchend' || event.type === 'touchcancel') {
+          // Clear the primary touch if it ended.
+          for (let i of event.changedTouches) {
+            // If our chosen primary touch ended, remove it.
+            if (GLFW.primaryTouchId === i.identifier) {
+              GLFW.primaryTouchId = null;
+              primaryChanged = true;
```
[the remark this answers is not in the document]
Comment (br0nk) [reply src/library_glfw.js:547]: Done

File: src/library_glfw.js [hunk 2]
```diff
@@ -490,11 +514,51 @@ var LibraryGLFW = {
...
+        let primaryChanged = false;
+
+        // Set a primary touch if we have none.
+        if (GLFW.primaryTouchId === null && event.type === 'touchstart' && event.targetTouches.length > 0) {
+          // Pick the first touch that started in the canvas and treat it as primary.
+          const chosenTouch = event.targetTouches[0];
+          GLFW.primaryTouchId = chosenTouch.identifier;
+
+          Browser.setMouseCoords(chosenTouch.pageX, chosenTouch.pageY);
+          Browser.setMouseCoords(chosenTouch.pageX, chosenTouch.pageY);
```
[the remark this answers is not in the document]
Comment (br0nk) [reply src/library_glfw.js]: Done

File: src/library_glfw.js [hunk 3]
```diff
@@ -435,7 +436,30 @@ var LibraryGLFW = {
...
+
+        // Don't let the browser fire a duplicate mouse event.
+        event.preventDefault();
+
+        let primaryChanged = false;
+        for (let i of event.changedTouches) {
+          // If our chosen primary touch moved, update Browser mouse coords
+          if (GLFW.primaryTouchId === i.identifier) {
+            Browser.setMouseCoords(i.pageX, i.pageY);
+            primaryChanged = true;
```
[the remark this answers is not in the document]
Comment (br0nk) [reply src/library_glfw.js:450]: Done

**Comment by br0nk** [comment]
Thanks for the speedy review.  Added [b6bd26d](https://github.com/emscripten-core/emscripten/pull/20805/commits/b6bd26d14444510d0c0e218bb9586c8cb3a545c4) addressing your comments.

Yes, this is intended to solely refactor the Browser object without changing the size or behavior of anything in library_browser.js.

I do, however, change the size of LibraryGLFW object in library_glfw.js.  Is there some tricky behavior here that w.r.t. the c bindings that I should be aware of?  As far as I can tell, there are no issues, but thought I should call it out.

**Comment by sbc100** [comment]
Yes, it makes sense that LibraryGLFW would change in size here.

**Review by sbc100 — APPROVED** [review]

**Comment by br0nk** [comment]
> Yes, this is intended to refactor the Browser object without changing the size or behavior of anything in library_browser.js.
> 

Just to clarify, this **does** add 2 methods to Browser, which shouldn't change the size of any instances, but may change the size of the object prototype.  From what I can tell this is fine here, but don't want to mess up some javascript Reflection magic that I'm not aware of.

**Pull request merged**